src/App.js

```diff
@@ -10,13 +10,35 @@ class App extends Component {
 
     render() {
         return (
-            <div className="App">
-                <input placeholder="city,country code" onChange={this.onCityChange}/>
-                <button onClick={this.onCitySubscribe}>Subscribe</button>
+            <div>
+                <div className="App">
+                    <input placeholder="city,country code" onChange={this.onCityChange}/>
+                    <button onClick={this.onCitySubscribe}>Subscribe</button>
+                    <button onClick={this.onPrint}>Print</button>
+                </div>
+                    {Object.entries(this.state.subscribedCities).map(([city, data]) => this.renderCityData(city, data))}
             </div>
         );
     }
 
+    onPrint = () => {
+        console.log(this.state.subscribedCities);
+    };
+
+    renderCityData = (city, data) => {
```

Comment:
👍 Nice function name.
The multiple `return` keywords are redundant, though:

```suggestion
    renderCityData = (city, data) => data.list.map(datum => (
      <div>{city}
        <div key={city + 'Date'}>Date: {datum.dt_txt}</div>
        <div key={city + 'temp'}>Temp: {(datum.main.temp - 273.15).toFixed(1)}ºC</div>
        <div key={city + 'humidity'}>Humidity: {datum.main.humidity}%</div>
        <div key={city + 'description'}>{datum.weather[0].description.toLocaleUpperCase()}</div>
        <div key={city + 'wind'}>Wind Speed: {(datum.wind.speed * 2.236936).toFixed(1)}mph</div>
      </div>
    );
```

Reply:
See above for how this function can be changed to just accept a single `data` parameter, and be able to extract the city name from `data.city.name`

Reply:
Absolutely agree with passing through a single parameter, I don't know why I didn't do it in the first place.